raster-foundry/raster-foundry
# Implement new histogram breakpoints and use for min/max

## Overview

Implement and use new draggable breakpoints for histograms

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

![image](https://user-images.githubusercontent.com/4392704/29433207-32058ec0-836c-11e7-807a-816edb3b7929.png)
![image](https://user-images.githubusercontent.com/4392704/29433214-38e5ebf4-836c-11e7-9522-f323bac55bc0.png)
![image](https://user-images.githubusercontent.com/4392704/29433217-3e6a43b8-836c-11e7-9cd7-c41a6c01d9d0.png)

## Notes
Clipping is broken on the tile server, so you won't be able to see changes on the histogram directly. Instead, you'll have to view the api requests and verify that they make sense.

## Testing Instructions

 * Test new breakpoints (used as min/max sliders in color correction)
* Verify that the api requests for updating color corrections are debounced while making changes
* Verify that changes persist as expected through the api
* Verify that changing the values using the input boxes works, and that the z-indexing of the inputs makes sense
* Verify that changing numbers with the keyboard makes sense:
  - Pressing enter applies number changes
  - Pressing up/down works, as does using the up/down arrows in the input box
  - Min/Max sliders can't switch places - instead they push the other one with it
  - Invalid numbers (too large, too small) are restricted to the correct range.
  - number precision specified is used

Closes #2388
